### ISSUE TITLE
Discord or Telegram to discuss Course Work

### ISSUE THREAD

**Issue by htmlHxcker** [javascript-tutorial/en.javascript.info#2718]
There may probably be one out there, but I think a Discord Server or Telegram Group would be a really nice addition to the course, not only would we be able to communicate with classmates and discuss solutions to the course, but then we would also be able to find study groups, I think it'll be a really useful addition.

PS: I don't know if there's one, I never saw any mention of any, but if there is, I'll be more than willing to join in.

**Comment by Manik2375** [comment]
Discord, yeah! I also think it will be useful

**Comment by iliakan** [comment]
Explain, please?

- Discourse?
- Discord?
- Telegram + bots?

Which one will be better and why?

**Comment by htmlHxcker** [comment]
I think Discord would be better because of the channels, that way it helps to keep discussions organized, e.g A Channel for assignment help or one for off-topic discussions so that way it's easier to find what you're looking for, I think there is a way to bridge Telegram and Discord but I'm not really sure about Discourse.

**Comment by iliakan** [comment]
I've never ever used Discord.

Only forums, including Discourse, and chats, such as Telegram et al.

But I heard Discord is cool overall.

How can we use it, what can it give to users? Please give some details ;)

P.S. Of course we can make bots and other integration stuff, whatever.

**Comment by htmlHxcker** [comment]
Okay, so Discord is like Slack but in a more relaxed way.
You have a server (Think Slack Workspaces) and that server is made up of channels. You can have two types of channels, audio, and text; and for each category, you can have as many channels as you want, usually, a channel is specific to one topic i.e 100 days of code or Exercises Help, the only difference is that for voice channels you can only do video and audio calls or streams, no text whatsoever whilst for Text channels, you can only have text, emojis GIFS, etc, no audio communication.

Why I'd prefer Discord over Discourse is because Discord is more suited for real-time communications than Discourse, then regarding Telegram, I think I'll prefer it just because of the ability to have multiple channels and keep the conversation clear take for example I want to talk about why I prefer to use React over Vue, I could talk about that in the off-topic channel and let the people talking about the Solution to the Recursion task talk about it in their channel, no stress.

Jeez at this rate I may as well join Discord's marketing team.
![screenshot-nimbus-capture-2021 09 15-16_02_01](https://user-images.githubusercontent.com/64586121/133458994-1503febe-c573-4114-9189-07be0d44b5f7.png)

**Comment by Manik2375** [comment]
Yeah, Discord have several benefits.
Such as: 

- Different channels for different Topics.
- Easy moderation.
- Can make bots easily with Discord.js
- Can assign different people with different powers (like moderators, contributors etc).
- Colorful codeblocks for sharing code
- etc..

Moderation and different channels make it more easy to handle, also there are number of amazing discord bots out there.

**Comment by iliakan** [comment]
What can we give to our users with Discord?
Besides a place to chat.

**Comment by iliakan** [comment]
P.S. Also, is discord fully free?

**Comment by Manik2375** [comment]
> P.S. Also, is discord fully free?

100% yes (There is also a feature called discord nitro for individual benefits, and others can boost your server for some server benefits)

> What can we give to our users with Discord?

A place to learn about things that are not in site? I mean a channel for js tips, articles etc. I would love it.

**Comment by iliakan** [comment]
So let's try.

I created Discord community server. How do I share it?

**Comment by htmlHxcker** [comment]
>What else should I set up?
Maybe a rules channel that has the rules of the Server and Code of Conduct (Some people can be extra rowdy), a channel for contributors and introductions, I'll advise that we don't have too many channels as that can make it fragmented.

**Comment by Manik2375** [comment]
Server making is almost done :+1:

**Comment by iliakan** [comment]
Should I put the link at the bottom of the site?
https://discord.gg/VSHd6hD6aN

**Comment by joaquinelio** [comment]
Don't be shy. It's a great idea. Draw a logo too. You NEED people there.

-
TLDR
It's a great idea. I wish I could commit to a SP version.

We at ES repo had -still have- a discord for the specific purpose of repo translation.
I was mad: "who we are", rooms, reunions, while the bot was enough to get things  under control and the "REAL" problems were unnatended: PR piling up and new tranlators taking files 4 years old

But it probed to be great. Rooms are less intimidating than github, people went there first and they were well guided by commited (not me, not enough I regret) maintainers.

The rooms are now empty. People enter, hang a question, leave.  

I comment this because you need the critical mass of people willing to help  each other in a reasonable time. Answers dont need to be inmediate but discord users may be used to the chat.

I guess the best kick start prospect are actual costudents: they share teacher, mates and specific study hours.

**Comment by htmlHxcker** [comment]
> Should I put the link at the title page, near Github?
> https://discord.gg/VSHd6hD6aN

I think you could add it there, I think what's left now is for us to get people in because in a way they're the lifeblood of it, and **we need to be watchful for raids.**

**Comment by Manik2375** [comment]
It's been added in site :+1: Now need to wait for people, it's lonely there currently :sweat_smile:

**Comment by iliakan** [comment]
I'll publish the notification on the site EU morning.

Should we install Giselle or another bot-server or do any changes at this moment, @htmlHxcker ?

**Comment by Manik2375** [comment]
I added some restrictions, like when someone get 2-3 warnings (warning will be automatic if someone use bad words, spam, too many mentions) he will get temporary mute automatically. 

We can add more restriction later when server will get some activity.

**Comment by iliakan** [comment]
🥳
